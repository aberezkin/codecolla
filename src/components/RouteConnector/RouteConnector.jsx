import React, {Component} from 'react';
import './RouteConnector.styl';

class RouteConnector extends Component {

    componentWillReceiveProps(newProps) {
        const { history } = this.props;

        if (!this.props.isReady && newProps.isReady) {
            this.props.connect(this.props.id);
            history.replace('/');
        }
    }

    render() {
        const { id } = this.props;

        return (
            <div className="route-connector">
                <h1>Connecting to peer: {id}</h1>
                <h1>spinner...</h1>
            </div>
        )
    }
}

export default RouteConnector;
