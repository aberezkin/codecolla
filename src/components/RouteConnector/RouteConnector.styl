.route-connector
  position absolute
  top 0
  left 0
  right 0
  bottom 0
  display flex
  flex-direction column
  justify-content center
  text-align center

