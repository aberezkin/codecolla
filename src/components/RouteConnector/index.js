import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import RouteConnector from './RouteConnector';
import { addPeerFromId } from '../../actions/index';

const RouteConnectorWithRouter = withRouter(RouteConnector);

const mapStateToProps = (state, ownProps) => ({
    isReady: state.peers.id !== '',
    id: ownProps.match.params.id,
});

const mapDispatchToProps = dispatch => ({
    connect: id => dispatch(addPeerFromId(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(RouteConnectorWithRouter);
