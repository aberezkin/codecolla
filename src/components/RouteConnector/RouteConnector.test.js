import React from 'react';
import RouteConnector from './RouteConnector';

const setup = propOverrides => {
    const props = {
        id: 'random-id',
        isReady: false,
        connect: jest.fn(),
        ...propOverrides,
    };

    const wrapper = shallow(<RouteConnector {...props}/>);

    return {
        props,
        wrapper
    }
};

describe('route connector', () => {
    it('should render connecting message', () => {
        const {wrapper} = setup();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correct peer id', () => {
        const {wrapper} = setup({id: 'other-id'});

        expect(wrapper).toMatchSnapshot();
    });
});