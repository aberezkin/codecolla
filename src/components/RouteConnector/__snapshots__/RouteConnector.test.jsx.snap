// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`route connector should render connecting message 1`] = `
<div
  className="route-connector"
>
  <h1>
    Connecting to peer: 
    random-id
  </h1>
  <h1>
    spinner...
  </h1>
</div>
`;

exports[`route connector should render correct peer id 1`] = `
<div
  className="route-connector"
>
  <h1>
    Connecting to peer: 
    other-id
  </h1>
  <h1>
    spinner...
  </h1>
</div>
`;
